homeassistant/components/hyperion/sensor.py

```diff
@@ -0,0 +1,189 @@
+from __future__ import annotations
+
+import functools
+from typing import Any
+
+from hyperion import client
+from hyperion.const import (
+    KEY_PRIORITIES,
+    KEY_UPDATE,
+    KEY_COMPONENTID,
+    KEY_VISIBLE,
+    KEY_OWNER,
+    KEY_ORIGIN,
+    KEY_PRIORITY
+)
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import slugify
+
+from . import (
+    get_hyperion_device_id,
+    get_hyperion_unique_id,
+    listen_for_instance_updates,
+)
+
+from .const import (
+    CONF_INSTANCE_CLIENTS,
+    DOMAIN,
+    HYPERION_MANUFACTURER_NAME,
+    HYPERION_MODEL_NAME,
+    SIGNAL_ENTITY_REMOVE,
+    TYPE_HYPERION_COMPONENT_SENSOR_BASE,
+    TYPE_HYPERION_SENSOR_VISIBLE_EFFECT,
+)
+
+def _sensor_unique_id(server_id: str, instance_num: int) -> str:
+    """Convert a sensor to a unique_id."""
+    return get_hyperion_unique_id(
+        server_id,
+        instance_num,
+        slugify(
+            f"{TYPE_HYPERION_COMPONENT_SENSOR_BASE} {TYPE_HYPERION_SENSOR_VISIBLE_EFFECT}"
+        ),
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Hyperion platform from config entry."""
+    entry_data = hass.data[DOMAIN][config_entry.entry_id]
+    server_id = config_entry.unique_id
+
+    @callback
+    def instance_add(instance_num: int, instance_name: str) -> None:
+        """Add entities for a new Hyperion instance."""
+        assert server_id
+        sensors = []
+        sensors.append(
+            HyperionSensor(
+                server_id,
+                instance_num,
+                instance_name,
+                entry_data[CONF_INSTANCE_CLIENTS][instance_num],
+                SensorEntityDescription(
+                    key="effect",
+                    translation_key="effect",
+                    icon="mdi:lava-lamp",
+                    entity_registry_enabled_default=True,
+                )                
+            ),
+        )
+
+        async_add_entities(sensors)
+
+    @callback
+    def instance_remove(instance_num: int) -> None:
+        """Remove entities for an old Hyperion instance."""
+        assert server_id
+        async_dispatcher_send(
+            hass,
+            SIGNAL_ENTITY_REMOVE.format(
+                _sensor_unique_id(server_id, instance_num),
+            ),
+        )
+
+    listen_for_instance_updates(hass, config_entry, instance_add, instance_remove)
+
+
+class HyperionSensor(SensorEntity):
```

Comment:
Since this is not a generic sensor, but rather one that reports the visible effect, I would suggest naming the class something more specific (e.g. `HyperionVisibleEffectSensor`) or perhaps have a base class with this name that is extended.